### PR TITLE
Budget-app

### PR DESCRIPTION
**Pull request by enis-memic**
## In this PR I did the following:
I created a Ruby on Rails application that allows the user to:
- register and log in, so that the data is private to them.
- introduce new transactions associated with a category.
- see the money spent on each category.
#### Interactions
- Splash screen
  - A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.

- Sign up and log in pages
  - The user should be able to register in the app with full name, email and password (all mandatory).
  - The user can log into the app using email and password.
  - If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

- Home page (categories page)
  - When the user logs in, they are presented with the categories page.
  - For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
  - When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
  - There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

- Transactions page
  - For a given category, the list of transactions is presented, ordered by the most recent.
  - At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
  - There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new category" page
  - The user fills out a form to create a new category, indicating their name and icon (both mandatory).
  - The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new transaction" page
  - The user fills out a form to create a new transaction with:
    - name (mandatory)
    - amount (mandatory)
    - categories (mandatory at least one)
  - The user click (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.
  - When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

#### Testing requirements
- Create unit and integration tests for all the most important components of your RoR application.

Video presentation - https://drive.google.com/file/d/13h7kNIQCQWb5sLXQPXssbrYjbA1pd0sW/view?usp=sharing

### CODE REVIEW

File: app/views/shared/_splash.html.erb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  <div>
+    <h1>Budget App</h1>
+  </div>
+  <div class="d-flex flex-column gap-2 mt-5">
+    <div>
+      <%= link_to "LOG IN", new_user_session_path, class:"btn btn-primary w-100" %>
+    </div>
+    <div>
+      <%= link_to "SIGN UP", new_user_registration_path, class:"btn btn-primary w-100" %>
+    </div>
```
Comment (elisha2kyakpo1) [inline app/views/shared/_splash.html.erb]: - [x] Great job creating a splash page, unfortunately, your design does not match the given design. I have attached images for your reference.

## Yours

yours

## design

og

File: app/views/shared/_categories.html.erb [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+          <% end %>
+        </div>
+      </div>
+    <% end %>
+  <% end %>
+</div>
+<%= link_to new_category_path,class:"nav-link" do %>
+  <section class="bg-success shadow text-center text-white p-2 mt-5">
+    <p class="border border-white my-3 p-1">Add New Category</p>
+  </section>
```
Comment (elisha2kyakpo1) [inline app/views/shared/_categories.html.erb]: - [ ] In order to match the given design, kindly tie this button to the bottom of the page and change the color to the one you have chosen for your project (only one color for the design).
- [ ] Kindly observe the background color and that of the cards, you will notice that there is a difference, also the typography.

original


category

Comment (elisha2kyakpo1) [inline app/views/shared/_categories.html.erb]: - [ ] Kindly tie the button to the bottom of the page and add the real color to it. The page background color and the cards look different. For this PR to be approved, all the requirements for the design given should be followed, such as colors, typography, and spaces between elements. I have attached an image for your reference.

original

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
Hi 👋 @enis-memic, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ The PR title and description are good and professional.
✔️ No linter errors.
✔️ Descriptive readme file.
✔️ Tests passing👏
👍 The app is deployed properly.
👏 The app works as expected.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/shared/_splash.html.erb (hunk 1 above)
Comment (enis-memic) [reply app/views/shared/_splash.html.erb]: @elisha2kyakpo1 Thank you for these suggestions!

File: app/views/shared/_categories.html.erb (hunk 2 above)
Comment (enis-memic) [reply app/views/shared/_categories.html.erb]: @elisha2kyakpo1 Thank you for these suggestions!

Comment (enis-memic) [reply app/views/shared/_categories.html.erb]: @elisha2kyakpo1 Thank you for these suggestions!

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @enis-memic, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights

✔️ Good communication
✔️ Nice UI
✔️ No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

 ## Required Changes ♻️
- [x] As the previous reviewer suggested Kindly make sure the `Add New Category` and `Add New Transaction` is tide to the bottom of the app, as for now there is a white space after the button

Screenshot_20230705_102241

_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by joel-kalema — APPROVED** [review]
Hi @enis-memic 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# Congratulations! 🎉 


## Highlights
Linters are passing 👍 
The project is running successfully 👍 
PR is well descriptive 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@joel-kalema** in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**